odin-dynamic/src/test/scala/com/permutive/logging/dynamic/odin/DynamicOdinLoggerSpec.scala

```diff
@@ -16,69 +16,157 @@
+  private implicit val arbPosition: Arbitrary[Position] =
+    Arbitrary(
+      Arbitrary
+        .arbitrary[(String, String, String, Int)]
+        .map((Position.apply _).tupled)
+    )
 
-  test("record a message") {
-    PropF.forAllF { (message: String) =>
-      val messages = runTest(_.info(message))
+  private val defaultConfig = RuntimeConfig(Level.Info)
 
-      messages.map(_.map(_.message.value).toList).assertEquals(List(message))
+  test("record only messages at the min level") {
+    PropF.forAllF { (debugMessage: String, infoMessage: String) =>
+      val messages = runTest()(logger =>
+        logger.debug(debugMessage) >>
+          logger.info(infoMessage)
+      )
+
+      messages
+        .map(_.map(_.message.value).toList)
+        .assertEquals(List(infoMessage))
     }
   }
 
-  test("update global log level") {
-    PropF.forAllF { (message1: String, message2: String) =>
-      val messages = runTest { logger =>
-        logger.info(message1) >> IO.sleep(10.millis) >> logger.update(
-          DynamicOdinConsoleLogger.RuntimeConfig(Level.Warn)
-        ) >> logger.info(
-          message2
+  test("disable logging for a particular enclosure") {
+    PropF.forAllF {
+      (
+          pos1Msg: String,
+          pos2Msg: String,
+          position1: Position
+      ) =>
+        val position2 =
+          position1.copy(enclosureName = position1.enclosureName + "2")
+        val messages = runTest(
+          RuntimeConfig(
+            Level.Info,
+            Map(
+              position2.enclosureName -> LevelConfig.Off
+            )
+          )
+        )(logger =>
+          logger.info(pos1Msg)(implicitly, position1) >>
+            logger.error(pos2Msg)(implicitly, position2)
         )
+
+        messages
+          .map(_.map(_.message.value).toList)
+          .assertEquals(List(pos1Msg))
+    }
+  }
+
+  test("raises default-level config") {
+    PropF.forAllF { (messageBeforeChange: String, messageAfterChange: String) =>
+      val messages = runTest() { logger =>
+        logger.info(messageBeforeChange) >>
+          IO.sleep(10.millis) >>
+          logger.update(RuntimeConfig(defaultLevel = Level.Warn)) >>
+          logger.info(messageAfterChange)
+      }
+      messages
+        .map(_.map(_.message.value).toList)
+        .assertEquals(List(messageBeforeChange))
+    }
+  }
+
+  test("lowers default-level config") {
+    PropF.forAllF { (messageBeforeChange: String, messageAfterChange: String) =>
+      val messages = runTest() { logger =>
+        logger.info(messageBeforeChange) >>
+          IO.sleep(10.millis) >>
+          logger.update(RuntimeConfig(defaultLevel = Level.Debug)) >>
+          logger.debug(messageAfterChange)
       }
-      messages.map(_.map(_.message.value).toList).assertEquals(List(message1))
+      messages
+        .map(_.map(_.message.value).toList)
+        .assertEquals(List(messageBeforeChange, messageAfterChange))
     }
```

Comment:
This would have failed in the old implementation as the logging level of `DynamicOdinConsoleLoggerImpl` would have been set to what ever the original value in `RuntimeConfig` was. This meant that you could raise the logging level, but never lower it!

The knock on effect of this change is covered in[ this comment](https://github.com/permutive-engineering/odin-contrib/pull/3/files#r1139244552)